django/db/models/expressions.py

```diff
@@ -866,6 +895,112 @@ def as_sql(self, compiler, connection, template=None, case_joiner=None, **extra_
+        clone.subquery.query.bump_prefix(query)
+
+        # Need to recursively resolve these.
+        def resolve_all(child):
+            if hasattr(child, 'children'):
+                [resolve_all(_child) for _child in child.children]
+            if hasattr(child, 'rhs'):
+                child.rhs = resolve(child.rhs)
+
+        def resolve(child):
+            if hasattr(child, 'resolve_expression'):
+                return child.resolve_expression(
+                    query=query, allow_joins=allow_joins, reuse=reuse, summarize=summarize,
+                    for_save=for_save)
+            return child
+
+        resolve_all(clone.subquery.query.where)
+
+        for key, value in clone.subquery.query.annotations.items():
+            if isinstance(value, SubQuery):
+                clone.subquery.query.annotations[key] = resolve(value)
+
+        return clone
+
+    def get_source_expressions(self):
+        return [x for x in [getattr(expr, 'lhs', None) for expr in self.subquery.query.where.children] if x]
+
+    def relabeled_clone(self, change_map):
+        clone = self.copy()
+        clone.subquery.query = clone.subquery.query.relabeled_clone(change_map)
+        clone.subquery.query.external_aliases.update(
+            alias for alias in change_map.values() if alias not in clone.subquery.query.tables)
+        return clone
+
+    def as_sql(self, compiler, connection, template=None, **extra_context):
+        connection.ops.check_expression_support(self)
+        template_params = self.extra.copy()
+        template_params.update(extra_context)
+
+        template_params['subquery'], sql_params = self.subquery.query.get_compiler(connection=connection).as_sql()
+
+        template = template or template_params.get('template', self.template)
+        sql = template % template_params
+        sql = connection.ops.unification_cast_sql(self.output_field) % sql
+        return sql, sql_params
+
+    def _prepare(self, output_field):
+        # If we are the rhs in a subquery, we want to remove the wrapping ()
+        if self.template == '(%(subquery)s)':
+            clone = self.copy()
+            clone.template = '%(subquery)s'
+            return clone
+
+        return self
+
+
+class Exists(SubQuery):
+    template = 'EXISTS(%(subquery)s)'
+
+    def __init__(self, *args, **kwargs):
+        self.negated = kwargs.pop('negated', False)
+        super(Exists, self).__init__(*args, **kwargs)
+
+    def __invert__(self):
+        return type(self)(self.subquery, self.output_field, negated=(not self.negated), **self.extra)
+
+    @property
+    def output_field(self):
+        return fields.BooleanField()
+
+    def resolve_expression(self, query=None, **kwargs):
+        # By definition, an EXISTS does not care about the columns, so the query can
+        # be simplified down to the primary key.
+        self.subquery = self.subquery.values('pk').order_by()
+        return super(Exists, self).resolve_expression(query, **kwargs)
```

Comment:
So, I think maybe we should remove the "automatic `.values('pk')`, and document that this is probably what you want to do; although I think leaving the `.order_by()` is reasonable, as it ensures that there is no ordering.

And we don't want ordering, because all that will do is force the database to sort, which will be slower.